### PR TITLE
Inject `global.*` values to make them available to release manager

### LINKED ISSUE #1066: Globally injected variables do not work in subcharts/dependencies
**Describe the bug**
In the MRO pipeline, Jenkins sets the parameter imageTag during the helm install. If the deployment.yaml is located in a dependent chart (charts/ folder) and contains `image: "{{ .Values.image.repository }}:{{ .Values.imageTag }}"`, helm cannot overwrite imageTag. This can only be achieved by accessing and setting the global parameter `image: "{{ .Values.image.repository }}:{{ .Values.global.imageTag }}"`; `--set global.imageTag=x.x.x`

**To Reproduce**
1. Clone Helm Chart Repo with nested chart structure
2. cd chart/
3. Access .Values.imageTag in deployment.yaml
4. Run `helm template .  -f values.yaml -f values.dev.yaml -f secrets.dev.yaml --set registry=image-registry.openshift-image-registry.svc:5000 --set componentId=<componentId> --set imageNamespace=<project>-dev --set imageTag=abc`
5. Check imageTag used for deployment
6. Use .Values.global.imageTag in deployment.yaml
7. Run `helm template .  -f values.yaml -f values.dev.yaml -f secrets.dev.yaml --set registry=image-registry.openshift-image-registry.svc:5000 --set componentId=<componentId> --set imageNamespace=<project>-dev --set global.imageTag=abc`
8. Check imageTag used for deployment

**Expected behavior**
imageTag should be set by MRO pipeline

**Screenshots**
-

**Affected version:**
 - OpenDevStack v4.x

### PR DESCRIPTION
**Pull request by serverhorror**
fix #1066

### CODE REVIEW

**Comment by metmajer** [comment]
Thanks @serverhorror and @dvvanessastoiber, we appreciate the fix. FYI @jafarre-bi @hrcornejo

**Comment by dvvanessastoiber** [comment]
> Doesn't seem to work enough, yet 😔
> 
> Release Manager wants the tag names to be present (the ones it creat s)
> 
> Any hints how those could be forwarded or overwritten so that Helm can receive the right values?

@serverhorror From what I saw in my tests, the generated image tag is correctly set and used. The only thing that is not working for me at the moment is the creation of the image tag in the **frontend** image stream in QA (backend is working).

**Comment by dvvanessastoiber** [comment]
> Let's take a look, in that case it might be the options used to build the image.
```
...
odsComponentStageBuildOpenShiftImage(
    context, [resourceName: "${context.componentId}-backend", imageTag: "v7.1.1", dockerDir: "docker/backend", buildTimeoutMinutes: 10]
  )
odsComponentStageBuildOpenShiftImage(
    context, [resourceName: "${context.componentId}-frontend", imageTag: "v7.1.1", dockerDir: "docker/frontend", buildTimeoutMinutes: 10]
  )
odsComponentStageRolloutOpenShiftDeployment(context, [
    imageTag: "v7.1.1", 
    deployTimeoutMinutes: 10, 
    selector: "app.kubernetes.io/instance=${context.componentId}",
    chartDir: 'chart',
    helmEnvBasedValuesFiles: [
      'values.env.yaml',
      'secrets.env.yaml'
    ],
    helmAdditionalFlags: ['--debug']
  ])
...
```
The options in the Jenkinsfile are basically the same for frontend and backend

**Comment by dvvanessastoiber** [comment]
@serverhorror I just wanted to confirm explicitly that this PR is fixing the underlying issue for me. Thank you!

**Comment by jafarre-bi** [comment]
Hello, @serverhorror @dvvanessastoiber thank you very much for the work you have done here. I think we should talk to clarify any open points. Will contact you offline.

**Comment by hrcornejo** [comment]
As was said in previous conversations here (and in bug #1066 ), the PR is not enough. The Release Manager needs the tags to be created to process images. Some examples:
We have this image created in DEV:
image
And after a release to QA, this is the image in the test environment:
image
The same image (SHA), but with a different tag.
In the example, a CM is using the tag created in DEV, not the one created in QA:
image

File: src/org/ods/component/HelmDeploymentStrategy.groovy [hunk 1]
```diff
@@ -126,9 +126,21 @@ class HelmDeploymentStrategy extends AbstractDeploymentStrategy {
                 // we persist the original ones set from outside - here we just add ours
                 Map mergedHelmValues = [:]
                 mergedHelmValues << options.helmValues
+
+                // we add the global ones - this allows usage in subcharts
+                // TODO: is this really needed? - we could use *just* the global ones
```
Comment (hrcornejo) [inline src/org/ods/component/HelmDeploymentStrategy.groovy:131]: Remove the TODOs. Thanks

**Review by hrcornejo — CHANGES_REQUESTED** [review]

File: src/org/ods/component/HelmDeploymentStrategy.groovy (hunk 1 above)
Comment (hrcornejo) [reply src/org/ods/component/HelmDeploymentStrategy.groovy]: Done

**Review by hrcornejo — APPROVED** [review]

**Pull request merged**